### PR TITLE
3887 Disable stemming and Synonyms on caseName searches

### PR DESCRIPTION
**Pull request by albertisfu**
According to #3887, this PR disables stemming on the `caseName` fields for RECAP, Opinions, and Oral Arguments.

This change applies to both the `case_name` filter and the text query.

To disable stemming, it is necessary to use the `caseName.exact` version. 

This ensures that both stemming and synonyms are disabled for the `case_name` filter and the text query on the `caseName` field.

If we'd like to disable only stemming and keep synonyms, it would require creating a different version of the `caseName` field where only stemming is disabled.

### CODE REVIEW

**Review by mlissner — APPROVED** [review]
LGTM. Let's get a second look, and see if I missed anything though. :)

File: cl/search/tests/tests_es_oral_arguments.py [hunk 1]
```diff
@@ -2189,7 +2189,8 @@ def test_stemming_disable_search(self) -> None:
         self.assertEqual(actual, expected)
         self.assertIn("Freedom of", r.content.decode())
         self.assertIn("<mark>Inform</mark>", r.content.decode())
-        self.assertEqual(r.content.decode().count("<mark>Inform</mark>"), 1)
+        print(" r.content.decode()", r.content.decode())
```
Comment (ERosendo) [inline cl/search/tests/tests_es_oral_arguments.py:2192]: This seems to have been added accidentally

File: cl/search/tests/tests_es_recap.py [hunk 2]
```diff
@@ -2698,6 +2698,109 @@ async def test_micro_cache_for_search_results(self, mock_fetch_es) -> None:
...
+        cd = {
+            "type": SEARCH_TYPES.RECAP,
+            "q": "Howell ind",
+        }
+        r = async_to_sync(self._test_article_count)(
+            cd, 0, "Disable stemming and synonyms"
+        )
+
+        de.docket.delete()
+        de_2.docket.delete()
```
Comment (ERosendo) [inline cl/search/tests/tests_es_recap.py:2802]: I believe we should wrap these transactions within the self.captureOnCommitCallbacks(execute=True) context manager to make sure Elasticsearch records are deleted.

**Comment by ERosendo** [comment]
I've reviewed the code and it looks good. However, I've been unable to reproduce the reported test failure, despite repeated attempts and variations in test execution conditions 🤔.

**Comment by albertisfu** [comment]
> I've reviewed the code and it looks good. However, I've been unable to reproduce the reported test failure, despite repeated attempts and variations in test execution conditions 🤔. 

Thanks @ERosendo  ! Yeah, this issue is not being caught locally. I’ve confirmed by adding a print statement that the text query is not working correctly in the tests on the CI. For some reason, it doesn't seem to be using the exact version of the field. I’ll continue debugging the problem.

File: cl/search/tests/tests_es_recap.py (hunk 2 above)
Comment (albertisfu) [reply cl/search/tests/tests_es_recap.py:2802]: I think `self.captureOnCommitCallbacks(execute=True)` is not required on tests for the delete signal. Due to `es_signal_processor` for RECAP, Opinions, and OA,  `transaction.on_commit` is not used to call `remove_document_from_es_index` within `handle_delete`?

File: cl/search/tests/tests_es_oral_arguments.py (hunk 1 above)
Comment (albertisfu) [reply cl/search/tests/tests_es_oral_arguments.py]: Thanks, removed!

File: cl/search/tests/tests_es_recap.py (hunk 2 above)
Comment (ERosendo) [reply cl/search/tests/tests_es_recap.py:2802]: oh you're right. it's not using `transaction.on_commit` 👍

**Comment by albertisfu** [comment]
The tests were correct and revealed the root issue. 

The reason they continued sometimes matching the stemmed version of `caseName` in the opinion search was that a test related to `build_more_like_this_query` sometimes was executed before. Within this method, `SEARCH_OPINION_QUERY_FIELDS` was being extended without being copied first. As a result, when it was used in a search, the stemmed version of `caseName` was included in the list of fields for search.

**Review by ERosendo — APPROVED** [review]

**Comment by mlissner** [comment]
Hell yeah, tests! Nice one, Alberto. Merged!

**Pull request merged**